app/handler/subjects.go

```diff
@@ -2,103 +2,111 @@ package handler
 
 import (
 	"encoding/json"
-	"errors"
 	"fmt"
 	"net/http"
 
+	"github.com/gorilla/mux"
 	gremcos "github.com/supplyon/gremcos"
-	"github.com/supplyon/gremcos/api"
 
+	"github.com/imeplusplus/dont-panic-api/app/dbOperations"
 	"github.com/imeplusplus/dont-panic-api/app/model"
 )
 
-func GetAllSubjects(cosmos gremcos.Cosmos, w http.ResponseWriter, r *http.Request) {
+func GetSubjects(cosmos gremcos.Cosmos, w http.ResponseWriter, r *http.Request) {
+	subjects, err := dbOperations.GetSubjects(cosmos)
 
-	g := api.NewGraph("g")
-	query := g.V()
-
-	res, err := cosmos.ExecuteQuery(query)
 	if err != nil {
-		fmt.Println("Failed to execute a gremling command")
-		//logger.Error().Err(err).Msg("Failed to execute a gremlin command")
+		fmt.Println(err)
+		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 
-	response := api.ResponseArray(res)
-	vertices, err := response.ToVertices()
-
-	if err == nil {
-		fmt.Println(vertices)
-	}
-
-	subjects := VerticesToSubjects(vertices)
-
 	w.Header().Add("Content-Type", "application/json")
 	if err := json.NewEncoder(w).Encode(subjects); err != nil {
 		fmt.Println(err)
 		w.WriteHeader(http.StatusInternalServerError)
 	}
 }
 
-func GetSubjectByName(cosmos gremcos.Cosmos, w http.ResponseWriter, r *http.Request) {
-	fmt.Println("Not implemented")
-}
+func GetSubject(cosmos gremcos.Cosmos, w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
 
-func UpdateSubject(cosmos gremcos.Cosmos, w http.ResponseWriter, r *http.Request) {
-	fmt.Println("Not implemented")
-}
+	subject, err := dbOperations.GetSubjectByName(cosmos, vars["name"])
 
-func DeleteSubject(cosmos gremcos.Cosmos, w http.ResponseWriter, r *http.Request) {
-	fmt.Println("Not implemented")
+	if err != nil {
+		fmt.Println(err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Add("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(subject); err != nil {
+		fmt.Println(err)
+		w.WriteHeader(http.StatusInternalServerError)
+	}
```

Comment:
This seems to be repeated on almost every function in this file. Maybe DRY.

Reply:
mas só repete o
```
		fmt.Println(err)
		w.WriteHeader(http.StatusInternalServerError)
```
Não sei se vale a pena trocar por uma função